### ISSUE TITLE
Cannot exclude compassView from VoiceOver (accessibility) when map is not north

### ISSUE THREAD

**Issue by krupupakku** [maplibre/maplibre-native#3058]
### MapLibre iOS Version

6.8.0

### iOS Version

18.0

### Device

iPhone 13

### What happened?

Hey! as I'm not using the compassView I need to exclude it from being read from the iOS accessibility VoiceOver but I noticed even if I'm setting
```
        map.compassView.isAccessibilityElement = false
        map.compassView.accessibilityElementsHidden = true
```

is the map is not facing north, it's being read by screen reader. It's also visible in accessibility list elements (two fingers, three taps on screen).
If the map is not rotated there's no problem, screen reader is ignoring the compass view

extra info: rendering or hiding compass view is not changing things.

Some screens:
| / | Map | List |
|--|--|--|
| Not North | <img src="https://github.com/user-attachments/assets/0f2bdf43-61b0-4360-84fb-28043282208e" width="200" /> | <img src="https://github.com/user-attachments/assets/cdccecc0-b2c8-4b49-8b3d-3917ab1d1e38" width="200" /> |
| North | <img src="https://github.com/user-attachments/assets/a06a9031-933c-431c-829b-8971b2379c59" width="200" /> | <img src="https://github.com/user-attachments/assets/68bd8365-370b-48b1-9738-d7396cf7197d" width="200" /> |

### Steps to reproduce

- just show a simple map, 
- try disable compassView from accessibility (I'm doing it in updateUIView)
- rotate the map to not face north

### Relevant log output

```shell

```

**Comment by louwers** [comment]
Thanks for the report. I am not super aware how Voice Over works.

We have a [developing guide](https://github.com/maplibre/maplibre-native/blob/main/platform/ios/CONTRIBUTING.md#create-the-xcode-project) if you want to try looking into this yourself.

https://github.com/maplibre/maplibre-native/blob/8b32c9a78309af3df8ffadf5fa4276a135212262/platform/ios/src/MLNCompassButton.mm#L4
